crates/client/src/client.rs

```diff
@@ -434,6 +434,7 @@ impl FuelClient {
     ) -> io::Result<(TransactionStatus, Option<Vec<Receipt>>)> {
         let tx_id = self.submit(tx).await?;
         let status = self.await_transaction_commit(&tx_id).await?;
+        // TODO: Remove `receipts` now that receipts are queryable from the `Success` or `Failure` status
         let receipts = self.receipts(&tx_id).await?;
```

Comment:
I removed this query and replaced it with a block to read the receipts from the transaction status retrieved earlier in this function. This introduces an extra `clone()` because we now duplicate the receipts. This is okay for now - in the future, we can remove this function entirely because it is now redundant, and the `submit_and_await_commit` endpoint will return receipts for all successful and failed transactions. I have added a `TODO` inline to highlight this.